### PR TITLE
BUG: use errors=replace for stdout

### PR DESCRIPTION
**Pull request by dnicolodi**

### CODE REVIEW

**Comment by dnicolodi** [comment]
Unfortunately the added test catches the issue reported in #535 only when run without pytest stdout capture (namely with the `-s` command line flag).

**Comment by ankith26** [comment]
I can confirm that this PR fixes my issue. Instead of an error and build fail now, I get question mark symbols (`?`) in stdout in place of the korean characters, and the build finishes correctly.

**Comment by dnicolodi** [comment]
> I can confirm that this PR fixes my issue. Instead of an error and build fail now, I get question mark symbols (`?`) in stdout in place of the korean characters, and the build finishes correctly.

Thanks for testing. We can choose other alternative representations for characters that cannot be represented in the current encoding. I went for the `?` because it is the simplest. Other options that make sense are 'backslashreplace' or 'namereplace'. https://docs.python.org/3/library/codecs.html#error-handlers I don't have an opinion about which one is the best choice.

**Comment by ankith26** [comment]
IMHO the `'replace'` error handling is fine

**Comment by rgommers** [comment]
> I went for the `?` because it is the simplest

+1 from me, seems like a good choice.

File: mesonpy/_util.py [hunk 1]
```diff
@@ -71,3 +71,20 @@ def update(self, description: str) -> None:
...
+    ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x04
+
+    kernel = windll.kernel32
+    stdout = kernel.GetStdHandle(STD_OUTPUT_HANDLE)
+    mode = DWORD()
+
+    if not kernel.GetConsoleMode(stdout, byref(mode)):
+        return False
+
+    return bool(kernel.SetConsoleMode(stdout, mode.value | ENABLE_VIRTUAL_TERMINAL_PROCESSING))
```
Comment (rgommers) [inline mesonpy/_util.py:90]: The one difference with `mesonbuild/mlog.py` is that that adds `or os.environ.get('ANSICON')` (see https://github.com/adoxa/ansicon/blob/37f92009f6ca4938a34a1d7cecff41d331c7423c/readme.txt#L119 for the docs of that env var). It's not clear to me that that case is covered by the checks before calling `setup_windows_console` here - should it be added?

**Review by rgommers — APPROVED** [review]
Looks great, thanks @dnicolodi. This looks fine to merge as is, and quite nice that we can drop a dependency. Just one question about the one difference with the corresponding code in Meson for color handling.

**Comment by dnicolodi** [comment]
I'm moving dropping the colorama dependency to another PR and merging this one.

File: mesonpy/_util.py (hunk 1 above)
Comment (dnicolodi) [reply mesonpy/_util.py]: I think it should be handled indeed. I had the impression that ansicon was some legacy thing, but I checked and the `ANSICON` environment variable is used by other terminal emulators on Windows. I'll add checking for the environment variable.

**Pull request merged**